### PR TITLE
new article for tab completion

### LINKED ISSUE #5094: Document .NET CLI Tab Completion
.NET CLI tab completion supported has been added to .NET Core 2.0 SDK, but we haven't documented it yet on docs.

Helpful link: https://github.com/dotnet/cli/blob/master/Documentation/general/tab-completion.md

Could be added to https://docs.microsoft.com/en-us/dotnet/core/tools/ or should this be a completely new topic?

/cc @KathleenDollard

### PR DESCRIPTION
**Pull request by Thraka**
## Summary

Added a new article detailing how to enable tab completion for various terminals and .NET Core 2.0+

Fixes #5094

### CODE REVIEW

File: docs/core/tools/enable-tab-autocomplete.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+:------------------------------------|:----------------------------------------------------------------------------|:--------------------------------
+`dotnet a⇥`                          | `dotnet add`                                                                 | `add` is the first subcommand, alphabetically.
+`dotnet add p⇥`                      | `dotnet add --help`                                                          | Tab completion matches substrings and `--help` comes first alphabetically.
+`dotnet add p⇥⇥`                    | `dotnet add package`                                                          | Pressing tab a second time brings up the next suggestion.      
+`dotnet add package Microsoft⇥`      | `dotnet add package Microsoft.ApplicationInsights.Web`                      | Results are returned alphabetically.
+`dotnet remove reference ⇥`          | `dotnet remove reference ..\..\src\OmniSharp.DotNet\OmniSharp.DotNet.csproj` | Tab completion is project file aware.
+
+## PowerShell
+
+To add tab completion to **PowerShell** for the .NET Core CLI, add the following code to your PowerShell profile stored in `$PROFILE`:
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]: how do you access this?
when I type notepad $PROFILE on PS, I get an error

File: docs/toc.md [hunk 2]
```diff
@@ -139,6 +139,7 @@
 ### [Extensibility Model](core/tools/extensibility.md)
 ### [Continuous Integration](core/tools/using-ci-with-cli.md)
 ### [Custom templates](core/tools/custom-templates.md)
+### [Enable CLI TAB completion](core/tools/enable-tab-autocomplete.md)
```
Comment (mairaw) [inline docs/toc.md]:
```suggestion
### [Enable TAB completion](core/tools/enable-tab-autocomplete.md)
```

File: docs/core/tools/enable-tab-autocomplete.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+---
+title: Enable tab completion .NET Core CLI
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]: Final title looks like Enable tab completion .NET Core CLI - .NET Core CLI. Perhaps remove .NET Core CLI from here?

File: docs/core/tools/enable-tab-autocomplete.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+description: This article teaches you how to enable tab completion for the .NET Core CLI in v2.0 and above for PowerShell, Bash, and zsh.
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]:
```suggestion
description: This article teaches you how to enable tab completion for the .NET Core CLI for PowerShell, Bash, and zsh.
```

File: docs/core/tools/enable-tab-autocomplete.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+migrate
+pack
+```
+
+If that command doesn't work, make sure that .NET Core 2.0 or above is installed. If it's installed, but that command still doesn't work, make sure that the `dotnet` command resolves to a version of .NET Core 2.0 and above.
+
+> [!TIP]
+> Use the `dotnet --version` command to see what version of `dotnet` your current path is resolving to. For more information, see [Select the .NET Core version to use](../versions/selection.md).
+
+
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]: remove extra empty line

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks for jumping on this one @Thraka. It has been long overdue. This looks great but I've left some comments.

File: docs/core/tools/enable-tab-autocomplete.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+title: Enable tab completion .NET Core CLI
+description: This article teaches you how to enable tab completion for the .NET Core CLI in v2.0 and above for PowerShell, Bash, and zsh.
+author: thraka
+ms.author: adegeo
+ms.date: 12/17/2018
+---
+
+# How to enable TAB completion for the .NET Core CLI
+
+Starting with .NET Core CLI v2.0, support for tab completion is included but disabled by default. This article describes how to enable tab completion for your terminal of choice.
```
[the remark this answers is not in the document]
Comment (Thraka) [reply docs/core/tools/enable-tab-autocomplete.md]: DOH, ok 😄

File: docs/core/tools/enable-tab-autocomplete.md (hunk 1 above)
Comment (Thraka) [reply docs/core/tools/enable-tab-autocomplete.md]: I'll edit this to say create/edit the profile stored in the $PROFILE variable. Your error is just due to the fact that the file doesn't exist unless you create it.

File: docs/core/tools/enable-tab-autocomplete.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+```powershell
+# PowerShell parameter completion shim for the dotnet CLI 
+Register-ArgumentCompleter -Native -CommandName dotnet -ScriptBlock {
+     param($commandName, $wordToComplete, $cursorPosition)
+         dotnet complete --position $cursorPosition "$wordToComplete" | ForEach-Object {
+            [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_)
+         }
+ }
+```
+
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]: I just tested this and I'd also add the following:
- Link to https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_profiles#the-profile-variable for more information about the $profile variable
- Note that you need to allow scripts to run and link to https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_execution_policies?#change-your-execution-policy for more info

File: docs/core/tools/enable-tab-autocomplete.md [hunk 8]
```diff
@@ -8,11 +8,11 @@ ms.date: 12/17/2018
 
 # How to enable TAB completion for the .NET Core CLI
 
-Starting with .NET Core CLI v2.0, support for tab completion is included but disabled by default. This article describes how to enable tab completion for your terminal of choice.
+Starting with .NET Core SDK v2.0, support for tab completion is included but disabled by default. This article describes how to enable tab completion for your terminal of choice.
```
Comment (KathleenDollard) [inline docs/core/tools/enable-tab-autocomplete.md]: It's not exactly disabled by default - the script you run to set it up varies by shell, and we do not have a good way to run shell specific stuff at install. 

"disabled by default" -> "needs to be set up for your terminal" 

Is it "terminal" or "shell"

**Comment by Thraka** [comment]
@KathleenDollard OK I used the word Shell (after researching it made more sense, thanks) and I rewrote the intro to be clearer. Please let me know if that works :)

@mairaw I linked to two other article sections in the `about_profile` PowerShell docs.

**Review by mairaw — APPROVED** [review]
:shipit:

File: docs/core/tools/enable-tab-autocomplete.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+title: Enable tab completion
+description: This article teaches you how to enable tab completion for the .NET Core CLI for PowerShell, Bash, and zsh.
+author: thraka
+ms.author: adegeo
+ms.date: 12/17/2018
+---
+
+# How to enable TAB completion for the .NET Core CLI
+
+Starting with .NET Core SDK v2.0, the .NET Core CLI supports tab completion. This article describes how to configure tab completion for three shells, PowerShell, Bash, and zsh. Other shells may have support for auto completion. Refer to their documentation on how to configure auto completion, the steps should be similar to the steps described in this article.
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]: Just noticed this 
```suggestion
Starting with .NET Core 2.0 SDK, the .NET Core CLI supports tab completion. This article describes how to configure tab completion for three shells, PowerShell, Bash, and zsh. Other shells may have support for auto completion. Refer to their documentation on how to configure auto completion, the steps should be similar to the steps described in this article.
```

File: docs/core/tools/enable-tab-autocomplete.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+```
+> dotnet complete "dotnet a"
+add
+clean
+--diagnostics
+migrate
+pack
+```
+
+If that command doesn't work, make sure that .NET Core 2.0 or above is installed. If it's installed, but that command still doesn't work, make sure that the `dotnet` command resolves to a version of .NET Core 2.0 and above. Use the `dotnet --version` command to see what version of `dotnet` your current path is resolving to. For more information, see [Select the .NET Core version to use](../versions/selection.md).
```
Comment (mairaw) [inline docs/core/tools/enable-tab-autocomplete.md]:
```suggestion
If that command doesn't work, make sure that .NET Core 2.0 SDK or above is installed. If it's installed, but that command still doesn't work, make sure that the `dotnet` command resolves to a version of .NET Core 2.0 and above. Use the `dotnet --version` command to see what version of `dotnet` your current path is resolving to. For more information, see [Select the .NET Core version to use](../versions/selection.md).
```

**Review by mairaw — COMMENTED** [review]
some last nits

File: docs/core/tools/enable-tab-autocomplete.md (hunk 9 above)
Comment (Thraka) [reply docs/core/tools/enable-tab-autocomplete.md]: Strange, I thought I committed that suggestion

**Pull request merged**